### ISSUE TITLE
ResolutionArguments.GetArgument<T>() fails if T is an Enum

### ISSUE THREAD

**Issue by MartinLorensen** [graphql-dotnet/graphql-dotnet#74]
```
      Connection<FooType>()
        .Name("foo")
        .Argument<EnumType<MyEnum>>("p", "p")
        .WithObject<Bar>()
        .Unidirectional()
        .Resolve(
          c => GetFoos(c.Object, c.GetArgument<MyEnum>("p"));
```

I had hoped c.GetArgument<MyEnum>("p")  would work, but it returns defailt(MyEnum), ignoring the actual value of p. If i use c.GetArgument<string>("p") I get the string. 

It seems the GetArgument<> method has all the references needed to get an instance of the underlying EnumerationGraphType (that knows the actual mapping from string to the enum value) for the named argument, so I guess it should be possible to resolve it - however it might complicate the GetArgument<> method a bit...

**Comment by MartinLorensen** [comment]
I noted almost any argument has the same issue. E.g.

 - A `StringGraphType`/`string `does not "unescape" e.g. backslashes.
 - A `GraphTypeList<GraphTypeString>` argument can only be accessed with `GetArgument<object[]>("arg").Cast<string>().ToList()`.

**Comment by tlil87** [comment]
Thanks for the report - will address this as well in my next PR.

**Comment by joemcbride** [comment]
With my latest changes, I am able to successfully get an enumeration argument.  This is using the `Argument<T>` off of the `ResolveFieldContext`.  I think we need to update the logic in `GetArgument<T>` on `ResolutionArguments` though.

**Comment by joemcbride** [comment]
@tlil87 One thing I notice in `GetArgument<T>` is that it is doing some DefaultValue checking.  The core engine also attempts to populate the `Arguments` collection if a value isn't provided.  I think that is duplicate logic.

**Comment by joemcbride** [comment]
This is where that happens: https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Execution/DocumentExecuter.cs#L323

**Comment by tlil87** [comment]
@joemcbride Yes, seems like duplicated logic now. I know there were a couple of cases that didn't work earlier (prior to recent work and improvements) where the default values weren't set, hence why I added it back then. Feel free to remove :+1:

**Comment by joemcbride** [comment]
This should be fixed in [0.9.2](https://github.com/graphql-dotnet/graphql-dotnet/releases/tag/v0.9.2).
